things/model/src/main/java/org/eclipse/ditto/things/model/signals/events/ThingMigrated.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.ditto.things.model.signals.events;
+
+import static org.eclipse.ditto.base.model.common.ConditionChecker.checkNotNull;
+
+import java.time.Instant;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+import org.eclipse.ditto.base.model.entity.metadata.Metadata;
+import org.eclipse.ditto.base.model.headers.DittoHeaders;
+import org.eclipse.ditto.base.model.json.FieldType;
+import org.eclipse.ditto.base.model.json.JsonParsableEvent;
+import org.eclipse.ditto.base.model.json.JsonSchemaVersion;
+import org.eclipse.ditto.base.model.signals.FeatureToggle;
+import org.eclipse.ditto.base.model.signals.UnsupportedSchemaVersionException;
+import org.eclipse.ditto.base.model.signals.commands.Command;
+import org.eclipse.ditto.base.model.signals.events.EventJsonDeserializer;
+import org.eclipse.ditto.json.JsonFactory;
+import org.eclipse.ditto.json.JsonField;
+import org.eclipse.ditto.json.JsonFieldDefinition;
+import org.eclipse.ditto.json.JsonObject;
+import org.eclipse.ditto.json.JsonObjectBuilder;
+import org.eclipse.ditto.json.JsonPointer;
+import org.eclipse.ditto.json.JsonValue;
+import org.eclipse.ditto.things.model.ThingId;
+
+/**
+ * This event is emitted after a {@link org.eclipse.ditto.things.model.Thing} was successfully migrated.
+ *
+ * @since 3.7.0
+ */
+@Immutable
+@JsonParsableEvent(name = ThingMigrated.NAME, typePrefix = ThingEvent.TYPE_PREFIX)
+public final class ThingMigrated extends AbstractThingEvent<ThingMigrated> implements ThingModifiedEvent<ThingMigrated> {
+
+    /**
+     * Name of the "Thing Migrated" event.
+     */
+    public static final String NAME = "thingMigrated";
+
+    /**
+     * Type of this event.
+     */
+    public static final String TYPE = TYPE_PREFIX + NAME;
+
+    private final ThingId thingId;
+    private final JsonPointer path;
+    private final JsonValue value;
+
+    private ThingMigrated(final ThingId thingId,
+            final JsonPointer path,
+            final JsonValue value,
```

Comment:
As far as I can see, this is always a `Thing`, right?
So why not use `Thing` type and only in the JSON representation convert to `JsonObject` to store it in JSON.